---
title: "Methods for analysis of spatial peptidergic communication networks"
author: "Donghyun Kim"
bibliography: refs.bib
---
::: {#fig-peptide-networks}
![](pepnet.png){width=800}

Peptidergic communication networks
:::

Communication in the brain can occur between pairs of cells that release neuropeptides and cells that sense these neuropeptides. Each pair constitutes a distinct communication network between neurons. 

In this study, we develop computational methods to study the organization of these networks.

### Introduction

Peptides released by one neuron can bind to G-protein coupled receptors (GPCRs) on other neurons and modulate their activity. Many such cognate neuropeptide- GPCR pairs have been identified in the literature. These constitute multiple molecularly distinct communication networks between neurons. Single cell transcriptomic data suggests that a vast majority of cortical neurons participate in such communication in a cell-type dependent manner [@smith2019peptidergic].

Spatial transcriptomic experiments measure gene expression while preserving context within the tissue. We consider a recent whole mouse brain dataset obtained with an in-situ hybridization-based technique called MERFISH [@yao2023wholebrain] and develop graph-based methods to study the organization of putative peptidergic communication networks.

We explore regimes in which neighborhood information is helpful for cell type classification with graph neural networks and adapt Node2Vec [@grover2016node2vec] to obtain multilayer graph embeddings. 


### Background on spatial transcriptomic data

Spatial transcriptomic experiments measure gene expression while preserving spatial context. 
There are many technologies to acquire such data, with trade-offs in spatial resolution, speed, and compatibility with other kinds of experiments on the same tissue sample [@lein2017promise, @marx2021spacetx, @ortiz2021spatial, @moffitt2022spacetxrev, @chen2023spacetxtech].
Some of these technologies, namely SeqFISH, MERFISH, Slide-Seq have been used to produce atlas scale datasets in the mouse brain by the BICCN~@hawrylycz2023guide.
Here we focus on a whole mouse brain dataset obtained with an in-situ hybridization based technique called MERFISH @yao2023wholebrain.
This dataset consists of expression measurements of approximately 500 genes in 4 million cells from a single mouse brain.

We focus on the primary visual cortex (VISp region) in this study as shown in @fig-spatial-data.

::: {#fig-spatial-data}
![](spatial.png){width=800}

VISp region of the brain. The different colors assigned to cells correspond to different subclasses.
:::

We note that the subclass distribution of cells in the dataset is imbalanced.

::: {#fig-subclass-dist}
![](subclass_dist.png){width=800}

Subclass distribution of cells in VISp
:::


### Background on neuropeptidergic communication

Ligands are small molecules that bind selectively to receptor proteins. Peptides are small chains of amino acids.

Here the ligands of interest are peptides, and the receptor proteins of interest are GPCRs [@rosenbaum2009GPCRreview].
Peptides released by one neuron can bind to GPCRs on other neurons and modulate their activity.

Thus, compatible (a.k.a cognate) peptide-GPCR pairs can be considered as unique communication channel   [@smith2019peptidergic]. 
Importantly, such peptidergic signaling is slower and longer lasting compared to synaptic signaling.

Combinatorial expression of neuropeptides and receptors enables local, type specific communication between neurons.
Computational studies also suggest that such neuron-type–specific local neuromodulation may be a critical piece of the biological credit-assignment puzzle [@liu2021credit_assignment].


### Neuropeptidergic communication as directed multilayer graphs 

Peptidergic communication networks can be viewed as directed multilayer graphs.

The multilayer graph G is the set $\{V, \{E_l\}\}$ where:

$V$ is the set of vertices (cells).

$E_l$ s the set of directed edges for the lth network.


For the $l$ th network, an edge exists from vertex $i$ to vertex $j$ if all the following conditions are met:

1. distance $d[i,j] \leq d_{threshold}$
2. vertex i expresses the peptide precursor gene
3. vertex j expresses the GPCR gene

We set $d_{threshold} = 40 \textmu m$ for all experiments unless mentioned otherwise. 

::: {.column-margin}
Code for graph construction can be found [here](https://github.com/donghyunkm/cciGraphs/blob/main/notebooks/1_exploring_data_graph.ipynb).
:::

::: {#fig-multilayer}
![](multilayer_graphs.png){width=800}

Multilayer graphs
:::

5 neuropeptide precursor and GPCR pairs were experimentally measured in the MERFISH dataset. The 5 pairs each make up a layer in the multilayer graph as shown in @fig-multilayer.


### Node classification with graph neural networks

We first explore regimes in which neighborhood information is helpful for cell type classification with graph neural networks.

::: {#fig-gnn}
![](gnn.png){width=500}

Graph attention network overview
:::

Graph neural networks combine features of nodes that are adjacent according to the underlying graph.

A nonlinear function ∅ of aggregated (sum here) weighted node features determines updated node representation.

In GATv2, the node features are weighted using a function of node pairs, $\theta$ [@brody2021attentive]. 

Functions $\phi$ and $\theta$ are learned to optimize a per-node classification objective.

::: {.column-margin}
Code for node classification with graph neural networks can be found [here](https://github.com/donghyunkm/cciGraphs/blob/main/notebooks/3_revisiting_gnn.ipynb).
:::

::: {#fig-gnn-results}
![](gnn_results.png){width=500}

Using GATv2 for supertype classification
:::

Here, we consider cells as nodes, and their gene expression as initial node features. Cognate neuropeptide-GPCR pairs that lie within the specified distance are connected with an edge. 

Cell type classification is challenging when few genes are available. We find that gene expression in neighboring cells can improve supertype classification in this regime.

### Node embeddings

Now that we found that neighboorhood information can be helpful for cell type classification, we wanted to see if the neighborhood structure of a cell by itself (without any gene expression information) can be informative of cell types. 

To do so, we use node embedding methods.

::: {#fig-node2vec}
![](node2vec.png){width=500}

Node2Vec overview
:::

**Node2Vec** learns an embedding $f(u)$ that is predictive of the neighborhood $\mathcal{N}(u)$ of each node $u$.

The neighborhood for each node is determined through fixed length, 2nd order random walks starting from each node.

Parameters $p$ and $q$ determine the transition probabilities for the random walk.

::: {#fig-spectral}
![](spectral.png){width=500}

Eigenvectors of the laplacian matrix
:::

**Spectral graph clustering** frames the problem as an eigen decomposition [@von2007tutorial]. Specifically, the eigenvectors of the Laplacian matrix are used as features for clustering.

### Simulation

We build intuition for how peptidergic connectivity could partition cells into meaningful groups even in the absence of explicit gene expression information with a toy problem.

::: {#fig-toy}
![](toy.png){width=700}

Overview of toy problem
:::

We start with a type x type adjaceny matrix that shows which cell types can communicate with which cell types. The type x type adjacency matrix can be sampled to obtain a cell x cell adjacency matrix. The block structure is equivalent to the graph having distinct connected components (as shown on the right).

::: {.column-margin}
Code for Node2Vec and spectral graph clustering applied to the simulation can be found [here](https://github.com/donghyunkm/cciGraphs/blob/main/notebooks/4_simulations.ipynb).
:::

::: {#fig-toy_emb}
![](tsne_2.png){width=300}

TSNE projection of Node2Vec embedding

:::

Clustering the embeddings obtained with Node2Vec is a scalable way to identify such components. We found that spectral graph clustering can identify these components too. However, spectral graph clustering has limited applicability to our dataset given its computational complexity of $O(n^3)$.       

Conencting this back to the problem of peptidergic communication, we can think of the dark blue, light blue, red, and green nodes as 4 cell types that communicate with each other, and the yellow, brown, orange, and violet nodes as 4 cell types that communicate with each other.

This type x type adjacency matrix denotes 1 communication network between the cells or 1 layer in the multilayer graph. With 1 layer, we can distinguish between 2 groups of cells.  

### Multilayer graph clustering

For multilayer graphs, one approach to obtain embeddings for the whole graph is to obtain independent embeddings for each layer in the multigraph. Combining information from each embedding can identify all the ‘types’ when the blocks are chosen carefully across layers.

If we have n types $(t_1,…,t_n)$ and k layers $(X_1,…, X_k)$:

$P(t_i) = \frac{1}{n}$

$P(t_i | X_1) = \frac{2}{n}$

$P(t_i | X_1, X_2) = \frac{4}{n}$

$P(t_i | X_1, X_2, ..., X_k) = 1$

where $k = \log_2 n$


::: {#fig-toy_2}
![](toy2.png){width=600}

An embedding is generated from each type x type matrix. The 3 embeddings are concatenated and visualized with TSNE.

:::

Node2Vec embeddings cluster according to the underlying block structure in a manner that is robust to sparsity as well as noise in the cell x cell graph.

Noise experiment             |  Sparsity experiment
:-------------------------:|:-------------------------:
![](noise.png)  |  ![](sparsity.png)



### Application to MERFISH VISp data

We inspect the utility of graph embeddings in identifying celltypes (subclass) and spatial groupings in a few different settings.

1. considering only proximity of cells in the graph (**distance**)
2. concatenated embeddings of individual peptidergic graphs (**independent**)
3. collapsing the multi-layer graphs into a single layer graph (**squashed**)
4. jointly considering the multilayer graphs with an **adapted** Node2Vec model (**joint**)

In the joint setting, we sample random walks from all layers to determine the neighborhood for each node (as opposed to neighborhoods constructed from only 1 layer). 

Note that the peptidergic multilayer graph does not contain any edges between layers.

::: {.column-margin}
Code for [distance](https://github.com/donghyunkm/cciGraphs/blob/main/notebooks/5a_application%20experiment_distance.ipynb), [independent](https://github.com/donghyunkm/cciGraphs/blob/main/notebooks/5b_application%20experiment_independent.ipynb), [squashed](https://github.com/donghyunkm/cciGraphs/blob/main/notebooks/5c_application%20experiment_squashed.ipynb), and [joint](https://github.com/donghyunkm/cciGraphs/blob/main/notebooks/5d_application%20experiment_joint.ipynb).
:::

::: {#fig-application}
![](application.png){width=600}

Topmost plots are TSNE projections of embeddings with colors obtained from leiden clustering. Bottom plots are projections of leiden clusters onto the VISp region. Accuracy refers to subclass classification accuracy using the embeddings. Note that classification accuracy using a random classifier is approximately 25%.
:::

We observe that even in the absence of explicit gene expression information, embeddings obtained from peptidergic connectivity contain information about cell types.  

We speculate that cell clusters obtained with leiden clustering correspond to spatial domains relevant from a functional point of view.


### Gene Imputation

The expression of peptidergic communication genes that were not part of the MERFISH dataset can be recovered through imputation.
Several tools have been benchmarked for this task [@li2022benchmarking], and we used the top performing model Tangram to impute values of all peptidergic communication genes. 

Tangram maps cells from a scRNA-seq dataset to cells (or voxels) from a spatial transcriptomic dataset [@biancalani2021deep]. 

$$
L(S, M) = \sum_{k}^{n_{genes}} \cos_{sim} ((M^T S)_{*, k}, G_{*,k}) 
$$

S: scRNA-seq matrix (cell x gene)

G: spatial transcriptomic data matrix (cell x gene)

M: mapping matrix (cell * cell)



Tangram learns a mapping matrix M defined such that $M_{ij} \geq 0$ is the probability of cell i (from scRNA-seq) mapping to cell $j$ (from spatial transcriptomic data). 

We assessed imputation performance by using peptidergic communication genes that were already measured as test data. 

::: {#fig-tangram}
![](tangram.png){width=600}

The diagrams on the left are scatter plots of measured and predicted gene expression values for test genes (colors indicate subclass). The diagrams on the right are violin plots of test gene expression for each subclass.
:::

We conclude that the correlation between predicted gene expression and actual gene expression is not strong enough for us to use predicted values for the 32 other peptidergic communication gene pairs. In particular, a non trivial amount of null gene expression values were mapped to positive values for most genes. 



### Limitations and future work

Only 5 out of the 37 cognate neuropeptide precursor and GPCR pairs were experimentally measured in the MERFISH dataset, which precludes an assessment at more detailed levels of the taxonomy 

Neuron morphology may be an important consideration relevant for the full spatial extent of peptidergic communication. Arbor density representations derived from patch-seq datasets could offer a way forward. 

\

The poster is attached below.

::: {#fig-poster}
![](Poster_draft.jpg){width=600}

Poster
:::

\

Code repository: <https://github.com/donghyunkm/cciGraphs>


\

We thank Tom Chartrand, Ian Convy, Olga Gliko, Anna Grim, Yeganeh Marghi, Uygar Sümbül, Meghan Turner, and Kasey Zhang for helpful feedback and discussions.